### PR TITLE
chore: nits from SA review

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

Nits from SA review. Focused on README, AnteHandler and tests and some of the internal keeper methods by stepping through with debugger.

Some are abstraction change suggestions which I'm happy to remove if folks disagree

### CODE REVIEW

File: x/smart-account/ante/ante_test.go [hunk 1]
```diff
@@ -187,13 +188,16 @@ func (s *AuthenticatorAnteSuite) TestSignatureVerificationWithAuthenticatorInSto
 }
 
 // TestSignatureVerificationOutOfGas tests that the ante handler exits early by running out of gas if the
-// fee payer has not been authenticated before consuming the 20_000 low gas limit (even if the specified limit is 300k)
+// fee payer has not been authenticated before consuming the parametrized max unauthenticated gas limit (even if the specified limit is 300k)
```
Comment (p0mvn) [inline x/smart-account/ante/ante_test.go:191]: Note: took me a while to process this test by referring to the hardcoded values (also I think 20k was wrong - it is 120k). Instead, changed to using the parameter value directly for clarity

File: x/smart-account/ante/circuit_breaker.go [hunk 2]
```diff
@@ -49,6 +49,9 @@ func (ad CircuitBreakerDecorator) AnteHandle(
 
 // IsCircuitBreakActive checks if smart account are active and if there is a
 // selected authenticator, the function will return true is the circuit breaker is active.
+// nit: this abstraction encapsulates more than just the circuit breaker - there is
+// also the check for the selected authenticator. Should we change it to
+// ShouldExecuteAuthenticatorFlow?
 func IsCircuitBreakActive(
```
Comment (p0mvn) [inline x/smart-account/ante/circuit_breaker.go:55]: Note: Please feel free to commit if disagreement. If PR is approved, I will change name pre-merge
```suggestion
func IsCircuitBreakActive(
```

File: x/smart-account/keeper/keeper.go [hunk 3]
```diff
@@ -156,6 +156,7 @@ func (k Keeper) GetInitializedAuthenticatorForAccount(
 const FirstAuthenticatorId = 1
 
 // GetNextAuthenticatorId returns the next authenticator id
+// nit: getter that mutates - should we change name to InitializeOrGetNextAuthenticatorId?
 func (k Keeper) GetNextAuthenticatorId(ctx sdk.Context) uint64 {
```
Comment (p0mvn) [inline x/smart-account/keeper/keeper.go:160]: Note: Please feel free to commit if disagreement. If PR is approved, I will change the name pre-merge
```suggestion
// nit: getter that mutates - should we change name to InitializeOrGetNextAuthenticatorId?
func (k Keeper) GetNextAuthenticatorId(ctx sdk.Context) uint64 {
```

File: x/smart-account/testutils/max_value_authenticator.go [hunk 4]
```diff
@@ -26,6 +26,7 @@ type MaxAmountAuthenticator struct {
 }
 
 func (m MaxAmountAuthenticator) StaticGas() uint64 {
+	// Q: I see it's only used in tests but is it safe to leave as 0?
 	return 0
```
Comment (p0mvn) [inline x/smart-account/testutils/max_value_authenticator.go:30]: Please commit if no concerns
```suggestion
	return 0
```

File: x/smart-account/ante/ante_test.go (hunk 1 above)
Comment (PaddyMc) [reply x/smart-account/ante/ante_test.go:191]: Yeah that's perfect thank you!

File: x/smart-account/ante/circuit_breaker.go (hunk 2 above)
Comment (PaddyMc) [reply x/smart-account/ante/circuit_breaker.go:55]: Here I'm a bit tentative to change it, I agree with the assessment, but am tentative to change function signatures rn. :bow:

File: x/smart-account/keeper/keeper.go (hunk 3 above)
Comment (PaddyMc) [reply x/smart-account/keeper/keeper.go:160]: This I completely agree with and I think we should change it

File: x/smart-account/ante/circuit_breaker.go (hunk 2 above)
Comment (p0mvn) [reply x/smart-account/ante/circuit_breaker.go:55]: Np. Let's close then

**Review by PaddyMc — APPROVED** [review]

**Comment by PaddyMc** [comment]
gg, sorry for the delay in reviewing

**Pull request merged**